db/migrate/20220202141917_add_columns_to_rules.rb

```diff
@@ -0,0 +1,9 @@
+class AddColumnsToRules < ActiveRecord::Migration[6.1]
+  def change
+    add_column :rules, :requires, :string
+    add_column :rules, :conflicts, :string
```

Comment:
We should not rely on `ref_id` when pointing at other entities, we have our own internal IDs for that purpose. Also, should not be one of these an array of items? Storing an array as a string is heavily inefficient, especially when we will use this to resolve conflicts and the lookups will be based on the array :thinking: 

So one of these fields should be an ID, the other one should be extracted into another table similarly to `profile_rules`. Also if you think about it, when you are internally using an ID, you should also store the type to reflect polymorphism.

https://guides.rubyonrails.org/association_basics.html#polymorphic-associations
https://guides.rubyonrails.org/association_basics.html#the-has-and-belongs-to-many-association

Reply:
Yeah, the reason I created the requires as a String is because I was thinking that instead of splitting the string into an array in the openscap parser, we could pass it as a string to compliane-backend and split it on our end

But I agree that it isn't a good idea to rely on the ref_ids.. I'll work on updating this

Reply:
Our end means before we store it in the DB :grin: